sdk/include/sdk/AsyncResult.h

```diff
@@ -59,6 +59,15 @@ struct VoidResult {};
 
 enum class CallState { ONGOING, CANCELING, COMPLETED, FAILED };
 
+/**
+ * @brief Status of an MQTT publish operation
```

Comment:
Pls keep it abstract of the underlying protocol (mqtt)
```suggestion
 * @brief Status of a publish operation
```